pkg/kubectl/resource_printer_test.go

```diff
@@ -1000,14 +1000,108 @@ func TestPrintPod(t *testing.T) {
-					Phase: "podPhase",
+					Phase: "Succeeded",
+					ContainerStatuses: []api.ContainerStatus{
+						{Ready: true, RestartCount: 3, State: api.ContainerState{Running: &api.ContainerStateRunning{}}},
+						{Ready: true, RestartCount: 3},
+					},
+				},
+			},
+			"",
+		},
+		{
+			// Test ready is not enough for reporting running
+			api.Pod{
+				ObjectMeta: api.ObjectMeta{Name: "test5"},
+				Spec:       api.PodSpec{Containers: make([]api.Container, 2)},
+				Status: api.PodStatus{
+					Reason: "OutOfDisk",
+					Phase:  "Failed",
+					ContainerStatuses: []api.ContainerStatus{
+						{Ready: true, RestartCount: 3, State: api.ContainerState{Running: &api.ContainerStateRunning{}}},
+						{Ready: true, RestartCount: 3},
+					},
+				},
+			},
+			"",
+		},
+	}
+
+	buf := bytes.NewBuffer([]byte{})
+	for _, test := range tests {
+		printPod(&test.pod, buf, false, false, false, []string{})
+		// We ignore time
+		if !strings.HasPrefix(buf.String(), test.expect) {
+			t.Fatalf("Expected: %s, got: %s", test.expect, buf.String())
+		}
+		buf.Reset()
+	}
+}
+
+func TestPrintAllPod(t *testing.T) {
+	tests := []struct {
+		pod    api.Pod
+		expect string
+	}{
+		{
+			// Test name, num of containers, restarts, container ready status
+			api.Pod{
+				ObjectMeta: api.ObjectMeta{Name: "test1"},
+				Spec:       api.PodSpec{Containers: make([]api.Container, 2)},
+				Status: api.PodStatus{
+					Phase: "Running",
+					ContainerStatuses: []api.ContainerStatus{
+						{Ready: true, RestartCount: 3, State: api.ContainerState{Running: &api.ContainerStateRunning{}}},
+						{RestartCount: 3},
+					},
+				},
+			},
+			"test1\t1/2\tRunning\t6\t",
+		},
+		{
+			// Test container error overwrites pod phase
+			api.Pod{
+				ObjectMeta: api.ObjectMeta{Name: "test2"},
+				Spec:       api.PodSpec{Containers: make([]api.Container, 2)},
+				Status: api.PodStatus{
+					Phase: "Pending",
+					ContainerStatuses: []api.ContainerStatus{
+						{Ready: true, RestartCount: 3, State: api.ContainerState{Running: &api.ContainerStateRunning{}}},
+						{State: api.ContainerState{Waiting: &api.ContainerStateWaiting{Reason: "ContainerWaitingReason"}}, RestartCount: 3},
+					},
+				},
+			},
+			"test2\t1/2\tContainerWaitingReason\t6\t",
+		},
+		{
+			// Test the same as the above but with Terminated state and the first container overwrites the rest
+			api.Pod{
+				ObjectMeta: api.ObjectMeta{Name: "test3"},
+				Spec:       api.PodSpec{Containers: make([]api.Container, 2)},
+				Status: api.PodStatus{
+					Phase: "Unknown",
+					ContainerStatuses: []api.ContainerStatus{
+						{State: api.ContainerState{Waiting: &api.ContainerStateWaiting{Reason: "ContainerWaitingReason"}}, RestartCount: 3},
+						{State: api.ContainerState{Terminated: &api.ContainerStateTerminated{Reason: "ContainerTerminatedReason"}}, RestartCount: 3},
+					},
+				},
+			},
+			"test3\t0/2\tContainerWaitingReason\t6\t",
+		},
+		{
+			// Test ready is not enough for reporting running
```

Comment:
Changed the comment. I decided not to modify the original `TestPrintPod()` and changed my test to `TestPrintNonTerminatedPod()`.